selfdrive/controls/controlsd.py

```diff
@@ -277,9 +275,10 @@ def state_control(frame, rcv_frame, plan, path_plan, CS, CP, state, events, v_cr
   angle_control_saturated = CP.steerControlType == car.CarParams.SteerControlType.angle and \
     abs(actuators.steerAngle - CS.steeringAngle) > STEER_ANGLE_SATURATION_THRESHOLD
 
+  saturated_count = saturated_count + 1 if angle_control_saturated and not CS.steeringPressed and active else 0
+
   # Send a "steering required alert" if saturation count has reached the limit
-  # TODO: add a minimum duration, now a warning will show every time the path changes briefly
-  if (active_count > SATURATION_TIMEOUT_ON_ENGAGE) and (lac_log.saturated or angle_control_saturated) and not CS.steeringPressed:
+  if (saturated_count > STEER_ANGLE_SATURATION_TIMEOUT) and (lac_log.saturated or angle_control_saturated) and not CS.steeringPressed:
```

Comment:
This now needs to be an `or`, otherwise it will never trigger.

```suggestion
  if (lac_log.saturated or angle_control_saturated or (saturated_count > STEER_ANGLE_SATURATION_TIMEOUT)) and not CS.steeringPressed:
```